add collection and filter for group_id

zisoft opened the pull request:
Closes #16651 

This PR adds a new collection and collection filter based on the `group_id`.

It only takes grouped images into account so the lists are not flooded with countless group_ids of ungrouped images.

Collections:
![Bildschirmfoto 2024-04-21 um 11 29 21](https://github.com/darktable-org/darktable/assets/4083281/257c76e7-75fc-4557-9e92-81878277dbc9)

Collection filters:
![Bildschirmfoto 2024-04-21 um 11 29 52](https://github.com/darktable-org/darktable/assets/4083281/aaa95684-4565-4838-8a87-fd7d37e045eb)

wpferguson commented:
YES! :tada: 

I tried this with a portrait shoot that i had done with ~1000 images.  I used the AutoGrouper script to group them into 20 groups, then narrowed down the collection using `group id`.  I just stepped through the groups by clicking.  Very Nice!

To make this accessible from Lua, please add the following line after line 4076 in src/libs/collect.c

`luaA_enum_value(L, dt_collection_properties_t, DT_COLLECTION_PROP_GROUP_ID);`

Thanks again @zisoft

wpferguson commented:
Here's a script to test the Lua change with.  Start it, then assign a keystroke to the shortcut.  Then hover over any image in a group and trigger the shortcut.  You should have only the group.  Trigger the shortcut again and it returns to the collection.

[groupinfo.zip](https://github.com/darktable-org/darktable/files/15052954/groupinfo.zip)

zisoft commented:
> To make this accessible from Lua, please add the following line after line 4076 in src/libs/collect.c

Done.

zisoft commented:
> `luaA_enum_value(L, dt_collection_properties_t, DT_COLLECTION_PROP_GROUP_ID);`

I assume this needs to be done for the recently added collections types as well?

- flash
- white balance
- exposure program
- metering mode

wpferguson commented:
Yes, please

zisoft commented:
Ok, done

wpferguson commented:
Thanks

zisoft commented:
I have no strong opinion here.
Unfortunately we don't have the possibility to give groups a name so when using the group id collection/filter we get presented a list of ids so I think `group id` isn't too bad.

From your proposals I would choose `group status` and `group`.

wpferguson commented:
I can't figure out a use case for grouping.  I can:
* select the group leaders from a collection with the definition of group leader being any image the the id and group id are  equal which results in a lot of groups of 1.  If I select a group leader (of a group of more than 1) and try to add group followers I get all the group followers in the database, not the group followers of the group leader.
* select the group followers which narrows the collection down to all the group followers, but if I try and use group leader to narrow it to just one group I find that group leader isn't an option.

IMO the purpose of the grouping options should be to return 1 or more complete groups of images and the grouping options (follower and leader) don't do that.

I think we should remove the current grouping options and just leave the group id and then rename that to group.  Or, rename group id to group and leave grouping for now but open an RFC issue seeking reasons to keep it.

wpferguson commented:
I traced grouping back to #1703.  It seems that collection grouping got added as part of fixing ratings not being applied to members of a group.  Perhaps he used it for debugging and it just got left in?

zisoft commented:
While implementing this PR I have played with the grouping filter as well and cannot imagine a use case for that filter either.

Very confusing: The filter shows the number of affected images in parenthesis. These numbers are always based on the selected base collection and don't change if that collection is narrowed down further by additional filters (i.e. rating).

zisoft commented:
- renamed collection type and filter: `group id` --> `group`
- removed the grouping collection type and filter

wpferguson commented:
Tested and works fine.  I also checked the assign rating to group to make sure that still worked too.

Looks good to me.

zisoft commented:
Release notes is also missing for #16479, we can combine them.

Added more collection types and filters for flash, white balance, exposure program, metering mode and image grouping.

Removed the old image grouping collection type and filter which gave confusing results.

Gendura commented:
Hi, 
I too have usecase where grouping was usefull :
I want to delete all useless RAW files (ie. not stared/developed). With grouping I can only target the ones being in a group and avoid deleting a RAW not having a follower.
Currently I'm trying to figure a way to do otherwise but I don't see how. Maybe with a lua script ?

_TBH one additional category usefull for my usecase would be "group of only clones" but the original filter is still very usefull._

wpferguson commented:
@Gendura could you open a new issue, preferably at https://github.com/darktable-org//lua-scripts/issues.  A lua-script is probably the best answer as it can handle more complex logic than a simple collection rule.  That will give us a place to hash this out without continuing on a closed PR.
